src/queries/role.ts

```diff
@@ -11,20 +13,37 @@ import type {
   QueryByQuery,
 } from './makeQueries';
 
+type Role = AccessibilityRole | TextMatch;
+
+type ByRoleOptions = {
+  name?: TextMatch;
+};
+
+const filterByAccessibleName = (
+  node: ReactTestInstance,
+  name: TextMatch | undefined
+) => {
+  if (!name) return true;
+
+  const { queryByText, queryByLabelText } = getQueriesForElement(node);
+  return Boolean(queryByText(name) || queryByLabelText(name));
```

Comment:
I think we should use the most permissive query here: `queryAll` as `query` might throw in case it finds two matching child elements:

```suggestion
  const nodeQueries = getQueriesForElement(node);
  return nodeQueries.queryAllByText(name).length > 0 || nodeQueries.queryAllByLabelText(name)).length > 0;
```

Reply:
Good catch!